apps/web/src/pages/Pools/PoolAdvertisementPage/components/LanguageRequirementDialog.tsx

```diff
@@ -0,0 +1,469 @@
+  },
+});
+
+type LanguageRequirementDialogProps = {
+  headingLevel?: HeadingProps["level"];
+};
+
+const LanguageRequirementDialog = ({
+  headingLevel,
+}: LanguageRequirementDialogProps) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const languageProfilesList = [
+    {
+      title: languageProfiles.frenchEssential,
+      description: languageProfileDescriptions.frenchEssential,
+    },
+    {
+      title: languageProfiles.englishEssential,
+      description: languageProfileDescriptions.englishEssential,
+    },
+    {
+      title: languageProfiles.bilingualIntermediate,
+      description: languageProfileDescriptions.bilingualIntermediate,
+    },
+    {
+      title: languageProfiles.bilingualAdvanced,
+      description: languageProfileDescriptions.bilingualAdvanced,
+    },
+    {
+      title: languageProfiles.various,
+      description: languageProfileDescriptions.various,
+    },
+  ];
+
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger>
+        <Button
+          mode="icon_only"
+          color="secondary"
+          icon={InformationCircleIcon}
+          aria-label={intl.formatMessage({
+            defaultMessage: "Learn about how language requirements work.",
+            id: "WhHVMt",
+            description:
+              "Info button label for pool language requirements details.",
+          })}
+        />
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>
+          {intl.formatMessage({
+            defaultMessage:
+              "Language requirements and second language proficiency",
+            id: "+7yxK0",
+            description: "Heading for the pool language requirements dialog",
+          })}
+        </Dialog.Header>
+        <Dialog.Body>
+          <div data-h2-display="base(grid)" data-h2-gap="base(x1)">
+            <div data-h2-display="base(grid)" data-h2-gap="base(x1)">
+              <div>
+                <Heading
+                  level={headingLevel}
+                  size="h6"
+                  data-h2-margin="base(0, 0, x.25, 0)"
+                >
+                  {intl.formatMessage({
+                    defaultMessage:
+                      "Language requirements on GC Digital Talent",
+                    id: "h1UJ1Q",
+                    description: "Sub-heading for language requirements dialog",
+                  })}
+                </Heading>
+                <p>
+                  {intl.formatMessage({
+                    defaultMessage:
+                      "These are the language profiles that you can find on GC Digital Talent",
+                    id: "sry9Wl",
+                    description:
+                      "Language profiles list for language requirements dialog",
+                  }) + intl.formatMessage(commonMessages.dividingColon)}
+                </p>
+              </div>
+              <ul data-h2-display="base(grid)" data-h2-gap="base(x.5)">
```

Comment:
I think it makes sense to use a `dl` here instead.

Reply:
Ya we could use that as well, seems to match up better semantically. I'm not sure how to style it like a `ul` though maybe we can alter the design. I can check in with @substrae.

Reply:
I can help you out here... we have that in [our component library](https://www.chromatic.com/component?appId=61e099a1bb1465003a73d3ce&csfId=ui-src-components-definitionlist&buildNumber=7501&k=664604abc7d608e858dbae60-1200px-interactive-true&h=7&b=-1) already :stuck_out_tongue:

Reply:
@substrae Is it okay if we change the built in unordered list design (currently in figma), to the built in [description list design](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)?
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/22059495/b682e235-b936-4ad9-bd13-8c0a02250b32)

Reply:
Hey @yonikid15 there's actually a design for the `<dl>` version in the Figma file. Gray asked me to mock it up last week or the week before.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/36efc36f-2c00-4760-93c5-aa1e566884c8)